### PR TITLE
Implementation of Expandable design for proprietary data exchange feature

### PR DESCRIPTION
**Pull request by AKalinich-Luxoft**
Fixes https://github.com/smartdevicelink/sdl_core/issues/1734

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
HMI behavior should be tested manually

### Summary
Made the following changes:
- Added new parameter "requestSubType" to OnSystemRequest notification from HMI
- Added new RequestType option in System Request view - "OEM_SPECIFIC"
- Added new field for entering and sending "requestSubType" value

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: ffw/BasicCommunicationRPC.js [hunk 1]
```diff
@@ -1149,6 +1149,14 @@ FFW.BasicCommunication = FFW.RPCObserver
         if (appID) {
           JSONMessage.params.appID = appID;
         }
+         if(subType && subType.length > 0){
```
[the remark this answers is not in the document]
Comment (AKalinich-Luxoft) [reply ffw/BasicCommunicationRPC.js]: @jacobkeeler fixed in 4428727

File: css/sdl.css [hunk 2]
```diff
@@ -827,6 +834,11 @@
     top: 180px;
 }
 
+#systemRequestView .requestSubTypeInput {
+    top: 217px;
+
```
[the remark this answers is not in the document]
Comment (AKalinich-Luxoft) [reply css/sdl.css]: @jacobkeeler fixed in 4428727

File: ffw/BasicCommunicationRPC.js [hunk 3]
```diff
@@ -1149,6 +1149,14 @@ FFW.BasicCommunication = FFW.RPCObserver
         if (appID) {
           JSONMessage.params.appID = appID;
         }
+        if (subType && subType.length > 0) {
+          if (type == 'OEM_SPECIFIC'||
+              type == 'HTTP'||
+              type == 'PROPRIETARY') {
```
Comment (jacobkeeler) [inline ffw/BasicCommunicationRPC.js]: Shouldn't need this restriction. The only cases where subType shouldn't be used are existing PTU Requests (`HTTP` request w/ `BINARY` fileType or `PROPRIETARY` request w/ `JSON` fileType): https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0083-Expandable-design-for-proprietary-data-exchange.md#impact-on-existing-code

Comment (AKalinich-Luxoft) [reply ffw/BasicCommunicationRPC.js]: @jacobkeeler good catch, thanks.
Fixed in https://github.com/smartdevicelink/sdl_hmi/pull/73/commits/300388142dec1b953bca9a00c2553c4009edbf5f

**Pull request merged**